### PR TITLE
Single component mode

### PR DESCRIPTION
**Pull request by scoheb**

### CODE REVIEW

File: pipelines/push-disk-images-to-cdn/push-disk-images-to-cdn.yaml
Comment (johnbieren) [inline pipelines/push-disk-images-to-cdn/push-disk-images-to-cdn.yaml:1]: This pipeline is missing the version bump and README change

**Review by johnbieren — COMMENTED** [review]
Passing the snapshot to apply-mapping when we already pass the snapshotSpec is quite gross. Can we just determine if we are single component mode in collect data and emit that as a task result? It would be better to just collect the entire snapshot in collect-data and then change every task to receive the snapshot instead of snapshotSpec and fetch the spec from that. But, that is a very large breaking change obviously

**Comment by scoheb** [comment]
> Passing the snapshot to apply-mapping when we already pass the snapshotSpec is quite gross. 

I am just passing the namespaced-name ... not the entire snapshot ... i.e `my-team-namespace-tenant/snapshotABC`

> Can we just determine if we are single component mode in collect data and emit that as a task result? It would be better to just collect the entire snapshot in collect-data and then change every task to receive the snapshot instead of snapshotSpec and fetch the spec from that. But, that is a very large breaking change obviously

how would this sound then?

1. examine snapshot labels in collect-data.
2. if single mode activated, emit result called `single-component = componentXYZ` (for example)
3. have apply-mapping accept new parameters called `single-component`
4. reduce the snapshot in apply-mapping (as already done) based on `single-component`

This removes the need to pass the namespaced snapshot to apply-mapping

**Comment by johnbieren** [comment]
> > Passing the snapshot to apply-mapping when we already pass the snapshotSpec is quite gross.
> 
> I am just passing the namespaced-name ... not the entire snapshot ... i.e `my-team-namespace-tenant/snapshotABC`
> 
> > Can we just determine if we are single component mode in collect data and emit that as a task result? It would be better to just collect the entire snapshot in collect-data and then change every task to receive the snapshot instead of snapshotSpec and fetch the spec from that. But, that is a very large breaking change obviously
> 
> how would this sound then?
> 
> 1. examine snapshot labels in collect-data.
> 2. if single mode activated, emit result called `single-component = componentXYZ` (for example)
> 3. have apply-mapping accept new parameters called `single-component`
> 4. reduce the snapshot in apply-mapping (as already done) based on `single-component`
> 
> This removes the need to pass the namespaced snapshot to apply-mapping

Yeah, that was the style I was attempting to suggest in my comment

File: pipelines/push-disk-images-to-cdn/push-disk-images-to-cdn.yaml
Comment (scoheb) [reply pipelines/push-disk-images-to-cdn/push-disk-images-to-cdn.yaml:1]: fixed

**Comment by johnbieren** [comment]
@scoheb this is in draft so I assume it isn't ready for review. If you move it to ready and I don't review in a day or so, can you ping me? Will likely mean I missed it

**Comment by scoheb** [comment]
this PR is going to be replaced with others...

**Pull request closed without merging**